src/tpm2_pytss/utils.py

```diff
@@ -71,6 +71,47 @@ def make_credential(
     return (credblob, secret)
 
 
+def credential_to_tools(
+    id_object: TPM2B_ID_OBJECT, encrypted_secret: TPM2B_ENCRYPTED_SECRET
+) -> bytes:
+    data = b""
+
+    # Add the header, consisting of the magic and the version.
+    data = data + int(0xBADCC0DE).to_bytes(4, "big") + int(1).to_bytes(4, "big")
+
+    # Add the id object and encrypted secret.
+    data = data + id_object.marshal()
+    data = data + encrypted_secret.marshal()
+
+    return data
+
+
+def tools_to_credential(
+    credential_blob: bytes,
+) -> Tuple[TPM2B_ID_OBJECT, TPM2B_ENCRYPTED_SECRET]:
+    magic = int.from_bytes(credential_blob[0:4], byteorder="big")
+    if magic != 0xBADCC0DE:
+        raise ValueError(f"bad magic, expected 0xBADCC0DE, got 0x{magic:X}")
+    version = int.from_bytes(credential_blob[4:8], byteorder="big")
+    if version != 1:
+        raise ValueError(f"bad version, expected 1, got {version}")
+
+    id_object_len = int.from_bytes(credential_blob[8:10], byteorder="big")
+    id_object = credential_blob[10 : 10 + id_object_len]
+
+    encrypted_secret_len = int.from_bytes(
+        credential_blob[10 + id_object_len : 12 + id_object_len], byteorder="big"
+    )
+    encrypted_secret = credential_blob[
+        12 + id_object_len : 12 + id_object_len + encrypted_secret_len
+    ]
+
+    return (
+        TPM2B_ID_OBJECT(credential=id_object),
+        TPM2B_ENCRYPTED_SECRET(secret=encrypted_secret),
+    )
```

Comment:
You can use the unmarshal method here for both TPM2B_ID_OBJECT, TPM2B_ENCRYPTED_SECRET, it returns a tuple, first part is the instance of the type, the other is how much data has been consumed by the unmarshaling.

Reply:
Done!